### PR TITLE
Configuration Generator GUI

### PR DESCRIPTION
**Pull request by OfekI**
We are students working on a class project and we decided to extend CheckStyle-IDEA to add a GUI for creating and modifying CheckStyle configuration files.
Our goal is to increase the productivity of creating and maintaining CheckStyle configurations, while reducing frustrations as well.
We are looking for feedback on our work and would greatly appreciate if you could answer the following questions for us.

1. Do you feel that this feature is useful and would be appreciated by users of the plugin?
1. Is there anything about the feature you feel is especially helpful? If so, what?
1. Is there anything you feel should not be included in this feature (that currently is)?
1. Is there anything major you feel is lacking from the feature (aside from supporting 3rd-party checks, we have plans to possibly add this in the future)?
1. Would you consider including this feature as part of the main distribution of CheckStyle-IDEA? If not, why not?

We have added to the README to include a User Manual for the Configuration Generator, although it is not on this branch. You can see it [here](https://github.com/s92025592025/checkstyle-idea/blob/master/README.md).

### CODE REVIEW

**Comment by jshiell** [comment]
Hi - I'm travelling for work at present; will have a look at this when I'm back later this week 👍

File: src/main/java/org/infernus/idea/checkstyle/ConfigGeneratorController.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  /** The model of the Checkstyle Configuration GUI, handles the data */
+  private final ConfigGeneratorModel model;
+  /** The view of the Checkstyle Configuration GUI */
+  private final ConfigGeneratorView view;
+
+  /**
+   * Instantiates a new ConfigGeneratorController by setting the categories in the
+   * view and adding the required listeners.
+   * 
+   * @param project The current IntelliJ IDEA project
```
Comment (izgzhen) [inline src/main/java/org/infernus/idea/checkstyle/ConfigGeneratorController.java]: is this argument obsolete?

**Comment by jshiell** [comment]
Hey,

As mentioned, am busy at work and busy for the next couple of weekends so having a look at this may take a couple of weeks - sorry!

Some of your questions I won't be able to answer until I've reviewed it, but this feature has been requested before, and I know it's supported in CS-Eclipse. So I think we can assume it's certainly desired by and of use to some users, although I've no idea as to how many.

My main question for inclusion in the main branch would be to how you plan to support this in the future? As you can probably tell from the recent change history I don't have a lot of time nor will to work on this at present, and that's been further limited by having support previously contributed features such as code-style import. So while I do agree this is a useful feature to some, and I'm impressed by the work you've put in, I would need some confidence that you don't just plan to contribute it and leave it be. I'd hate to accept it and then have to remove it later due to a lack of support.

Another question, perhaps a variant - have you considered releasing the configuration editor as its own plugin? It does seem a nice self-contained tool, and perhaps it would be better to have it as a standalone plugin - given you control over code, releases, extension, and support - and instead look at introducing some features here to simplify bouncing configurations between the two plugins? As above I'm open to merging it, but it's perhaps worth some thought on this option anyway.

**Comment by tsjensen** [comment]
The fact that custom checks are not currently supported is a show stopper imho, because many enterprise installations use their own custom checks, and we don't want to break things for them. Also, we know from the Eclipse-CS editor that such a feature takes a lot of effort to maintain.
So, while it's generally really cool to have, and it's great to see these guys having put in this kind of work:  
They haven't responded to your question for two months, so I wouldn't be overly optimistic about their ability to support such a major functional element over a prolonged period of time.
So my 2 cents recommendation is to close this PR here, and hope they'll manage to put this out as a plugin of their own. I'd be one of its first users!

**Comment by jshiell** [comment]
Given the lack of response, I'm not confident this feature would be maintained. As such I'll close this based on current information.

**Pull request closed without merging**